eclipse-collections-api/src/main/java/org/eclipse/collections/api/stack/ImmutableStack.java

```diff
@@ -162,6 +162,15 @@
         return this.asLazy().<P, V>collectWith(function, parameter).toBag().toImmutable();
     }
 
+    /**
+     * @since 10.0.0
+     */
+    @Override
+    default <V> ImmutableBag<V> countByEach(Function<? super T, ? extends Iterable<V>> function)
+    {
+        return this.asLazy().<V>flatCollect(function).toBag().toImmutable();
```

Comment:
I think all the usages of generic types specified on methods are redundant. See 8e14f7f60.

Reply:
Does the static analysis actually show a warning and IntelliJ does not show a compiler error? It should be a compiler error if just <V> is getting inferred.
flatCollect() has expanded scopes, so it will return of type <? extends V> and toBag() and toImmutable() will cascade to <? extends V>, we have to force the scope to <V>. So I think the generic types are necessary.

Reply:
The generic types are not necessary. I created that commit to show that Travis still passes. IntelliJ warns about the redundant types. Presumably, the TeamCity build would fail too.

Reply:
IntelliJ does warn about redundant types. I see below that some TeamCity builds have failed too. So, should the generic types be removed?

Reply:
Yes please!

Reply:
I have removed the generics.